Patch from Antti P Miettinen to fix small optimization problem with
newer gcc's on arm. bz#322806
--- a/mozilla/xpcom/reflect/xptcall/src/md/unix/xptcinvoke_arm.cpp
+++ b/mozilla/xpcom/reflect/xptcall/src/md/unix/xptcinvoke_arm.cpp
@@ -212,7 +212,7 @@ XPTC_InvokeByIndex(nsISupports* that, PR
     "add	sp, sp, r4	\n\t"	/* restore stack pointer		*/
     "mov	%0, r0		\n\t"	/* the result...			*/
     : "=r" (result)
-    : "r" (&my_params)
+    : "r" (&my_params), "m" (my_params)
     : "r0", "r1", "r2", "r3", "r4", "ip", "lr", "sp"
     );
     
